dandiapi/zarr/views/__init__.py

```diff
@@ -49,7 +51,35 @@ class Meta:
         ]
         fields = ['name', 'dandiset', *read_only_fields]
 
-    dandiset = serializers.RegexField(f'^{Dandiset.IDENTIFIER_REGEX}$')
+    dandiset = serializers.PrimaryKeyRelatedField(
+        queryset=Dandiset.objects.all(),
+        pk_field=DandisetIdentifierField(),
+    )
```

Comment:
Does this automatically raise an exception if the specified dandiset ID is not found? And if so, what's the response code (seems like it might be 400 instead of 404)?